### PR TITLE
PSP-8114 : Expose Mayan errors to frontend (temporarily)

### PR DESCRIPTION
**Pull request by eddherrera**

### CODE REVIEW

File: source/backend/api/Helpers/Middleware/ErrorHandlingMiddleware.cs [hunk 1]
```diff
@@ -130,6 +130,13 @@ private async Task HandleExceptionAsync(HttpContext context, Exception ex)
 
                 _logger.LogWarning(ex, ex.Message);
             }
+            else if (ex is MayanRepositoryException)
```
Comment (devinleighsmith) [inline source/backend/api/Helpers/Middleware/ErrorHandlingMiddleware.cs:133]: Am I missing something? I see that you've added a new MayanRepositoryException class, but I don't see that being thrown anywhere.

File: source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs [hunk 2]
```diff
@@ -49,18 +50,24 @@ public async Task<ExternalResponse<DocumentTypeModel>> TryCreateDocumentTypeAsyn
...
             content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
 
             Uri endpoint = new($"{_config.BaseUri}/document_types/");
 
             var response = await PostAsync<DocumentTypeModel>(endpoint, content, authenticationToken);
+            if (response.Status == ExternalResponseStatus.Error)
+            {
+                _logger.LogDebug("Error while Creating document type");
+
+                if (_config.ExposeErrors.HasValue && _config.ExposeErrors.Value)
```
Comment (devinleighsmith) [inline source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs:64]: does config.ExposeErrors == true work?

File: source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs [hunk 3]
```diff
@@ -71,18 +78,26 @@ public async Task<ExternalResponse<DocumentTypeModel>> TryUpdateDocumentTypeAsyn
...
 
             Uri endpoint = new($"{_config.BaseUri}/document_types/{documentType.Id}/");
 
             var response = await PutAsync<DocumentTypeModel>(endpoint, content, authenticationToken);
 
+            if(response.Status == ExternalResponseStatus.Error)
+            {
+                _logger.LogDebug("Error while updating a document type");
+
+                if (_config.ExposeErrors.HasValue && _config.ExposeErrors.Value)
```
Comment (devinleighsmith) [inline source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs:93]: how would you feel about internalizing this behaviour into the mayan exception itself?

File: source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs (hunk 2 above)
Comment (eddherrera) [reply source/backend/api/Repositories/Mayan/MayanDocumentRepository.cs]: Updated

**Review by devinleighsmith — APPROVED** [review]

**Pull request merged**